core/src/main/java/hudson/model/View.java

```diff
@@ -1000,6 +1002,18 @@ public synchronized void doDoDelete(StaplerRequest req, StaplerResponse rsp) thr
      */
     public abstract Item doCreateItem( StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException;
 
+    /**
+     * An API method to get the allowed {$link TopLevelItem}s and its categories.
+     *
+     * @return A {@link Categories} entity that is shown as  JSON file.
+     *
+     * @since TODO
+     */
+    @Restricted(DoNotUse.class)
+    public Categories doCategories(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
+        return ItemGroupMixIn.getCategories(Jenkins.getAuthentication(), Jenkins.getInstance());
```

Comment:
@jglick Since I can invoke the service from `jenkins` object (means `/`)... Why not?